File: backend/app/schemas/category.py
from pydantic import BaseModel


class CategoryCreate(BaseModel):
    name: str
    color: str = "#0052CC"
    icon: str = "📁"
    is_personal: bool = False


class CategoryUpdate(BaseModel):
    name: str | None = None
    color: str | None = None
    icon: str | None = None
    position: int | None = None


class CategoryResponse(BaseModel):
    id: int
    position: int
    name: str
    color: str
    icon: str
    is_personal: bool

    model_config = {"from_attributes": True}

File: ios/Familienkalender/Models/Category.swift
import Foundation

struct CategoryCreate: Codable {
    let name: String
    var color: String = "#0052CC"
    var icon: String = "📁"
}

struct CategoryUpdate: Codable {
    var name: String?
    var color: String?
    var icon: String?
}

struct CategoryResponse: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let color: String
    let icon: String
}
